README.md

```diff
@@ -47,43 +67,132 @@ $ npm run start:dev
+
+### Tests application in this project
+- Unit tests - Logic validation with mocks;
+- Component tests - Validation with resources from docker images, using fixture resources to build the enviroment like:
+  queues, nosql databases and relational databases;
+- Integration tests - Validation of integration with staging enviroment (write and read operations only).
+
+### Diagram to exemplify the general idea
+![Test Strategies](docs/test-strategies.png)
+
+## Customizing tests details
+To Obtain info about Jest coverage configuration look [here](https://jestjs.io/pt-BR/docs/configuration#coveragereporters-arraystring--string-options).
+to get some information about ReportOptions look [here](https://use-form.netlify.app/interfaces/_node_modules__types_istanbul_reports_index_d_.reportoptions.html). 
+
+## Running test
+To run the unit tests of the project you can execute the follow command:
+
+### Unit tests
+Execute the follow command:
+```bash
+npm run test
+```
+
+### Component tests
+Start the docker containers:
+```bash
+./bin/runenv.sh
+```
+
+Booting the environment:
+```bash
+./bin/boot.sh
+```
+
+Executing the tests:
+```bash
+npm run test:component
+```
 
-### Response
-```json
-{
-  "alive": true,
-  "applicationName": "Core APIs Nest Service Template",
-  "grpcPort": "5000",
-  "restPort": "3001",
-  "env": "development",
-  "message": "Uncle Bob we love you"
-}
+### Integration tests:
+Copy the file `integration.example.env` to `integration.env` and edit it with de staging parameters.
+
+Executing the tests:
+```bash
+npm run test:integration
+```
+### End-to-End tests
+```bash
+npm run test:e2e
+```
+### Generating coverage reports
+To execute coverage tests you can execute the follow commands:
+#### Unit test coverage
+Execute the follow command:
+```bash
+npm run test:cov
 ```
 
-## Test
+#### Component test coverage
+Execute the follow command:
+```bash
+npm run test:component-cov
+```
 
+#### Integration test coverage
+Execute the follow command:
 ```bash
-# unit tests
-$ npm run test
+npm run test:integration-cov
+```
+
+> Observation:
+> The result can be found in the folder `target/*`.
+
 
-# e2e tests
-$ npm run test:e2e
+## License
+See the license [LICENSE.md](LICENSE.md).
 
-# test coverage
-$ npm run test:cov
-``` 
+## Contributions
+- [Guilherme M. Lopes](https://github.com/GuilhermeMLS)
+- [Gabriel Miranda](https://github.com/gabe-miranda)
+- [Angelo Weiss](https://github.com/FerraBraiZ)
+- [Pedro Fellipe](https://github.com/PedroFellipe) 
+- [Anderson de Oliveira Contreira](https://github.com/andersoncontreira)
 
 ## Getting started with boilerplate
 
 Download and extract the project, then:
 ```bash
-$ mv nest-service-template-production {your_project_name}
-$ git init
-$ git remote add origin https://github.com/{user}/{repo}.git
+mv skeleton-nest-microservice-api {your_project_name}
```

Comment:
I would call the project `nestjs-service-template`. It is simple and pretty accurate.